fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt

```diff
@@ -30,7 +39,24 @@ import org.mozilla.fenix.theme.FirefoxTheme
 @Composable
 fun DebugOverlay() {
     val snackbarState = remember { SnackbarHostState() }
-    val scope = rememberCoroutineScope()
+    val drawerState = remember { DrawerState(initialValue = DrawerValue.Closed) }
+    var drawerEnabled by remember { mutableStateOf(false) }
```

Comment:
I'm wondering if we need both of these states or if we could just reuse one. It seems like their intent is roughly the same

Reply:
I'll see if they can be consolidated

Reply:
Oh, the separate boolean was because of [this comment above](https://github.com/mozilla-mobile/firefox-android/pull/4798#discussion_r1423019686). 

Essentially, I was seeing a little animation weirdness when telling the drawer UI to be added back to the tree AND then open itself (on the very first open). In order to solve this, I added the boolean and the launched effect around it, so the boolean would be updated, the UI would be added back to the tree, and then in a coroutine (the launched effect) the drawer would be told to open.

Reply:
That makes sense, but maybe a quick comment to that effect for future readers?